deps/media-playback/media-playback/decode.c

```diff
@@ -338,15 +338,12 @@ static int decode_packet(struct mp_decode *d, int *got_frame)
 		}
 
 		int err = av_hwframe_transfer_data(d->sw_frame, d->hw_frame, 0);
+		if (err == 0) {
+			err = av_frame_copy_props(d->sw_frame, d->hw_frame);
+		}
 		if (err) {
```

Comment:
These should be mutually exclusive, so I think they can be part of the same `if-else`.

| condition  | err | evaluation |
| ---------- | --- | ---------- |
| `err == 0` | `0` | `true`     |
| `err == 0` | `1` | `false`    |
| `err`      | `0` | `false`    |
| `err`      | `1` | `true`     |

Reply:
The point is that `av_frame_copy_props()` may set `err` to a non-zero error code as well.

Reply:
Ah, you are correct. Sorry about that. My mind was elsewhere. Disregard.